src/settings.js

```diff
@@ -2,44 +2,76 @@ import { isPlainObject } from './helpers';
 import { Value } from './values';
 
 export default class Settings {
-  static prefix;
-
-  constructor(settings) {
-    this.cache = new Map();
-    this.proxy = new Proxy(settings, {
-      get: (target, key) => {
-        if (this.cache.has(key)) {
-          return this.cache.get(key);
-        }
+  static assign(...args) {
+    return new Settings(
+      Object.assign(
+        ...args.map(arg => {
+          if (arg instanceof Settings) {
+            const settings = arg._settings;
+            Object.entries(settings).forEach(([key, value]) => {
+              if (value instanceof Value && value.options.envPrefix === undefined) {
+                settings[key].options.envPrefix = arg._prefix;
+              }
+            });
```

Comment:
The inner bit of this might be nice as a helper or method on Settings. `settings.toJS()`, maybe?